common/compositor/va/varenderer.cpp

```diff
@@ -42,11 +43,27 @@ VARenderer::~VARenderer() {
   }
 }
 
+static uint32_t GetScalingModeFromEnv() {
+  char value[PROPERTY_VALUE_MAX];
+  property_get("board.hwc.scaling.mode", value, "0");
+  switch (atoi(value)) {
+    case 0:
+      return VA_FILTER_SCALING_DEFAULT;
+    case 1:
+      return VA_FILTER_SCALING_FAST;
+    case 2:
+      return VA_FILTER_SCALING_HQ;
+    default:
+      return VA_FILTER_SCALING_DEFAULT;
+  }
+  return VA_FILTER_SCALING_DEFAULT;
+}
 bool VARenderer::Init(int gpu_fd) {
 #ifdef ANDROID
   unsigned int native_display = ANDROID_DISPLAY_HANDLE;
   va_display_ = vaGetDisplay(&native_display);
   UNUSED(gpu_fd);
+  filter_flags_ = GetScalingModeFromEnv();
```

Comment:
no need get every time, just get at startup

Reply:
@xuguangxin , Yes, thie Init just do when startup. But not every time.